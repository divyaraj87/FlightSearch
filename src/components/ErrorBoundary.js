import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';

export class ErrorBoundary extends Component {
    render() {
        const { todos: { error } } = this.props;
        return (
            <div class="error">
            {error}
            </div>
        );
    }
}

ErrorBoundary.prototypes = {
}

export const mapStateToProps = (state) => ({
    todos: state.todos
});

export default connect(mapStateToProps,
    {})(ErrorBoundary);