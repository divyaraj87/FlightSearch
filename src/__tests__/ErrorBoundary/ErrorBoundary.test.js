import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import  configureMockStore from 'redux-mock-store';
import  ErrorBoundary   from '../../components/ErrorBoundary';

const dispatchSelectChangeSpy = sinon.spy();
const dispatchProgressChangeSpy = sinon.spy();
const todos = {
  error: 'error'
};
const error= 'error';
let mockProps = {
    bgcolor: '#000',
    completed: 10,
    width: 10,
    error: 'error',
    todos: {
      error: 'error'
    }
};
const mockStore = configureMockStore();
const store = mockStore({});
describe('ErrorBoundary', () => {
  let pageWrapper;

  describe('ErrorBoundary Page', () => {
    const originalProps = { ...mockProps };

    beforeEach(() => {
      pageWrapper = shallow( <Provider>  
        <ErrorBoundary />   </Provider>    
      );
    });

    afterEach(() => {
      mockProps = { ...originalProps };
    });

    it('should render ErrorBoundary', () => {
      expect(pageWrapper).toMatchSnapshot();
    });
  });
});
