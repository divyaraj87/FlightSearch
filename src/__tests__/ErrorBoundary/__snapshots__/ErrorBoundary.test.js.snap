// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary ErrorBoundary Page should render ErrorBoundary 1`] = `undefined`;
